lib/modules/manager/npm/extract/common/dependency.ts

```diff
@@ -181,8 +181,22 @@ export function extractDependency(
     dep.versioning = npmVersioningId;
     dep.packageName = githubOwnerRepo;
   } else {
-    dep.skipReason = 'unversioned-reference';
-    return dep;
+    // <protocol>://[<user>[:<password>]@]<hostname>[:<port>][:][/]<path>[#<commit-ish> | #semver:<semver>]
+    // https://docs.npmjs.com/cli/v10/configuring-npm/package-json#git-urls-as-dependencies
+    const len = 7; // length of 'semver:'
+    const maybeVersion = depRefPart.substring(len);
+
+    if (depRefPart.startsWith('semver:') && isValid(maybeVersion)) {
+      dep.currentRawValue = dep.currentValue;
```

Comment:
what is `currenRawValue` ? 🤔

I've never seen it before.

@rarkins do you know something? or is it totally wrong?

Reply:
Well, it is assigned in all other cases as well. That's why I copied it.

Reply:
Yeah, let's keep it - it's used